logdetective/server/database/models/metrics.py

```diff
@@ -97,6 +106,47 @@ def update(
             metrics.response_certainty = response_certainty
             session.add(metrics)
 
+    def add_mr(
+        self,
+        mr_id: int,
+        project_id: datetime,
+        job_id: int,
+    ) -> None:
+        """This request was triggered by a merge request.
+        Link it."""
+        mr = MergeRequests.get_or_create(mr_id, project_id, job_id)
+        self.merge_request_id = mr.id
+        with transaction(commit=True) as session:
+            session.merge(self)
+
+    @classmethod
+    def get_requests_metrics_for_mr(
+        cls,
+        mr_id: int,
+        project_id: int,
+        job_id: int,
+    ) -> Optional["AnalyzeRequestMetrics"]:
```

Comment:
I'm not sure what this means. Does it return `AnalyzeRequestMetrics` only sometimes, what is returned in other cases if not? Line 146 would indicate that result should be a list of some kind.

Reply:
good catch, it should be a list.